workspaces/src/rpc/client.rs

```diff
@@ -47,7 +49,7 @@ impl Client {
         near_jsonrpc_primitives::types::transactions::RpcTransactionError,
     > {
         retry(|| async {
-            let result = JsonRpcClient::connect(&self.rpc_addr).call(method).await;
+            let result = self.rpc_client.call(method).await;
```

Comment:
I think I have played around with this in the past, but I was getting https://github.com/hyperium/hyper/issues/2112 at runtime. If examples and tests work for you with this change I might have been doing something wrong.

Reply:
yeah, everything works on my end, possibly it was an old version of jsonrpc-client? cc @miraclx 

I can dig more into this tomorrow to see what's happening internally, but if there is doubt, this shouldn't come in 0.2 likely

Reply:
similar reason why I kept it as `connect` since it was erroring out if I tried reusing the client back then

Reply:
Yeah, this should've worked right from time. The likely errors from early versions of the jsonrpc-client probably would've been from `JsonRpcClient::call` taking `self` instead of `&self` that would've allowed you to trivially reuse the client for multiple RPC method calls. Technically, you could've safely `clone`d the client back then since it used an `Arc` underneath, but it wasn't very clear from the type system.

Then again, multiple calls to `connect` is not so bad, since the internal `Client` is reused across multiple `connect` calls.

It's useful, however, to share a single `RpcClient` instance, which saves on extra heap allocations needed to store the server address and request headers.